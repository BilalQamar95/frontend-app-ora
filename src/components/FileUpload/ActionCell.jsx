import React, { useCallback } from 'react';
import { IconButton, Icon } from '@edx/paragon';

import { useIntl } from '@edx/frontend-platform/i18n';
import { Delete, Preview } from '@edx/paragon/icons';

import messages from './messages';

const ActionCell = ({
  onDeletedFile,
  disabled,
  row,
}) => {
  const { formatMessage } = useIntl();
  const deleteFile = useCallback(async () => {
    console.log('deleteFile', row.index);
    await onDeletedFile(row.index);
  }, []);
  return (
    <>
      <IconButton
        src={Delete}
        alt={formatMessage(messages.deleteButtonAltText)}
        iconAs={Icon}
        onClick={deleteFile}
        disabled={disabled}
      />
      <IconButton
        src={Preview}
        alt={formatMessage(messages.previewButtonAltText)}
        iconAs={Icon}
        disabled={disabled}
      />
    </>
  );
};

export default ActionCell;
