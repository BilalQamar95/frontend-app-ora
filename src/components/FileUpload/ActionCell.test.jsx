import { shallow } from '@edx/react-unit-test-utils';
import ActionCell from './ActionCell';

describe('<ActionCell />', () => {
  const props = {
    onDeletedFile: jest.fn(),
    disabled: false,
    row: {
      index: 0,
    },
  };
  it('renders', () => {
    const wrapper = shallow(<ActionCell {...props} />);
    expect(wrapper.snapshot).toMatchSnapshot();
  });
});
