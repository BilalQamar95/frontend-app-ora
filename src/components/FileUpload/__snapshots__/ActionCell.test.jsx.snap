// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionCell /> renders 1`] = `
<Fragment>
  <IconButton
    alt="Delete"
    disabled={false}
    iconAs="Icon"
    onClick={[Function]}
    src={[Function]}
  />
  <IconButton
    alt="Preview"
    disabled={false}
    iconAs="Icon"
    src={[Function]}
  />
</Fragment>
`;
